File: basic-proxy-impl/src/main/java/com/study/proxy/impl/util/ClassNameProvider.java
package com.study.proxy.impl.util;

public class ClassNameProvider {

    private static final String SIMPLE_NAME = "$SimpleProxy";

    public String className() {
        return internalName().replace('/', '.');
    }

    public String internalName() {
        return "com/study/proxy/impl/" + SIMPLE_NAME;
    }

    public String typeDescriptor() {
        return String.format("L%s;", internalName());
    }

    public String simpleName() {
        return SIMPLE_NAME;
    }

}

File: basic-proxy-kotlin-impl/src/main/kotlin/com/study/proxy/impl/util/ClassNameProvider.kt
package com.study.proxy.impl.util

object ClassNameProvider {

    const val SIMPLE_NAME = $$"$SimpleProxy"

    fun className() = internalName().replace('/', '.')

    fun internalName() = "com/study/proxy/impl/$SIMPLE_NAME"

    fun typeDescriptor() = "L${internalName()};"

}
